control/config.go

```diff
@@ -84,6 +84,49 @@ func GetDefaultConfig() *Config {
 	}
 }
 
+// UnmarshalJSON unmarshals valid json into a Config.  An example Config can be found
+// at github.com/intelsdi-x/snap/blob/master/examples/configs/snap-config-sample.json
+func (c *Config) UnmarshalJSON(data []byte) error {
```

Comment:
Added the same comment to all of those sections for all 5 files being modified; hopefully that wording is clear enough